### Table of Cotents
1. [February 1 - 5]  (#february-1st-to-5th)
2. [February 8 - 12]  (#february-8th-to-12th)
3. [February 15 - 19]  (#february-15th-to-19th)
4. [February 22 - 26]  (#february-22nd-to-26th)
## February 1st to 5th

1. Bulk segregant analysis using maize RNA-seq data [previously published](http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0036406), is used to test sdm based transcript arrangement to find/ narrow down *glossy* allele


## February 8th to 12th

1. Bulk segregant analysis using maize RNA-seq - continued the data analysis and made a list of candidate genes, using different levels of stringency of quality filtering and it influence on final list

2. Update the code to make use of polyploid variant data in snp density method


## February 15th to 19th

1. Analayzing RNA-seq data from tetrapolid wheat and modifying code to deal with polypolidy and variants

2. Started Assembly of Barley exome data from [Mascher et al., Genome Biology, 2014] (http://genomebiology.biomedcentral.com/articles/10.1186/gb-2014-15-6-r78)

3. Release 29 of Hordeum vulgare genome donwloaded from [emsemblegenomes] (ftp://ftp.ensemblgenomes.org/pub/plants/release-29/fasta/hordeum_vulgare/dna/Hordeum_vulgare.082214v1.29.dna.genome.fa.gz)

4. Release 29 ref sequence was used to call variant analysis using both wt and mnd mutant bulks and variant information is available in the [mnd_data folder] (../002_snps_in_assembled_genome/barley/Mascher_GB2014/barley_snp_data/mnd_data)

5. Release 29 has both anchored data for 7 chromosome sequences and unordered and un anchored scaffold seqeunces in total of 19,705 sequences.

6. Using [ordered_fasta_vcf_positions.rb] (../002_snps_in_assembled_genome/barley/Mascher_GB2014/barley_snp_data/ordered_fasta_vcf_positions.rb) script varint position distribution and ratio distribution in different bin szies was plotted to [all scaffolds] (../002_snps_in_assembled_genome/barley/Mascher_GB2014/barley_snp_data/mnd_data/all_scaffolds) and only [anchored chromosome seqeuences] (../002_snps_in_assembled_genome/barley/Mascher_GB2014/barley_snp_data/mnd_data/only_chromosomes)

7. From the variant data on chromosome seqeunces only, we could see a peak of hm/ht ratio around 2.75 gigabases, whcih does indicate the causative location of Chromosome 5.

## February 22nd to 26th

1. Udpated sdm code to select variants with allele frequency of 0.75 or higher and changes to code that discard variants from fragment data. Repeat analysis to imrpove the selection of variants possibly hosting mutation.

2. Comparing bam data for both mutatnt and backgrond bulks to discard that variants that have more than 33% representation in background bulks. Since we are looking at recessive mutations, variants which have more than 33% representation in bacground mulk may not be candidate mutations (based on segregation patterns of 1:2:1 - out 2 heterozygotes 1 will make up recessive allele making 1 in 3 showing wildtype phenotype).

3. Density plots of the analysis are improved

