### Table of Cotents
1. [March 1 - 12]  (#march-1st-to-12th)
2. [March 14 - 18]  (#march-14th-to-18th)
3. [March 21 - 25]  (#march-21th-to-25th)

## March 1st to 12th

1. Added code to calculate polyploidy proportions from pileup data and updated methods to take only fragments with variants

2. Prepared for my department seminar.


## March 15th to 19th

1. Analayzing RNA-seq data from tetrapolid wheat and modifying code to deal with polypolidy and variants

2. Started Assembly of Barley exome data from [Mascher et al., Genome Biology, 2014] (http://genomebiology.biomedcentral.com/articles/10.1186/gb-2014-15-6-r78)

3. Release 29 of Hordeum vulgare genome donwloaded from [emsemblegenomes] (ftp://ftp.ensemblgenomes.org/pub/plants/release-29/fasta/hordeum_vulgare/dna/Hordeum_vulgare.082214v1.29.dna.genome.fa.gz)

4. Release 29 ref sequence was used to call variant analysis using both wt and mnd mutant bulks and variant information is available in the [mnd_data folder] (../002_snps_in_assembled_genome/barley/Mascher_GB2014/barley_snp_data/mnd_data)

5. Release 29 has both anchored data for 7 chromosome sequences and unordered and un anchored scaffold seqeunces in total of 19,705 sequences.

6. Using [ordered_fasta_vcf_positions.rb] (../002_snps_in_assembled_genome/barley/Mascher_GB2014/barley_snp_data/ordered_fasta_vcf_positions.rb) script varint position distribution and ratio distribution in different bin szies was plotted to [all scaffolds] (../002_snps_in_assembled_genome/barley/Mascher_GB2014/barley_snp_data/mnd_data/all_scaffolds) and only [anchored chromosome seqeuences] (../002_snps_in_assembled_genome/barley/Mascher_GB2014/barley_snp_data/mnd_data/only_chromosomes)

7. From the variant data on chromosome seqeunces only, we could see a peak of hm/ht ratio around 2.75 gigabases, whcih does indicate the causative location of Chromosome 5.