## July 2016



1. Based on published work about [the recombination landscape in Arabidopsis thaliana F2 populations] (http://www.nature.com/hdy/journal/v108/n4/full/hdy201195a.html), I have tried to implement observations made about number of cross overs per plant and frequency distrubtuon in pools.


2. Modelled information is used to simulate segregating F2 populations and generate bulk fasta sequences and [the code is available here] (https://github.com/shyamrallapalli/simulate_bulk_sequence)

3. Resulting bulks of mutant and background phenotype fasta squences can be used as inputs for High throughput sequence generation as a population level sequence with each invidudal in the bulk with equal probability.

4. Used [GemSim software](http://bmcgenomics.biomedcentral.com/articles/10.1186/1471-2164-13-74) to generate Illumina type paired end sequencing data.


5. Cheripic provides result as text file with selected variants. So made R-shiny apps that takes the selected variants text file and plots density distribution.
The app also allows users to filter the region of interest, interactively, to limit the number of variants for downstream analysis. Code for the shiny app is [available on github](https://github.com/shyamrallapalli/cheripic-shiny-app)

