---
output:
  html_document:
    fig_caption: yes
    theme: spacelab
    toc: no
---
# **pacbio sequence read lengths - arabidopsis**

Load libraries and the data
```{r setup, include=FALSE}
# load the required libraries and functions
library(knitr)
library(markdown)
#opts_chunk$set(dev = 'pdf')
library(grid)
library(gridExtra)
library(ggplot2)
library(MASS)

## cusotom ggplot theme edits
mytheme <- theme(panel.background = element_rect(fill='white', colour='black'),
        text = element_text(size = 20, face='bold', colour='black'),
        axis.title.x = element_text(size = 19, face='bold', color='black', vjust = -1),
        axis.text = element_text(size = 17, face='bold', color='black'),
        panel.grid.major = element_line(linetype = 'dotted', colour='black'),
        panel.border = element_rect(fill=NA, linetype = "solid", colour='black'),
        plot.margin=unit(c(2,2,5,2),'mm'), legend.position=c(0.2,0.1), legend.title = element_blank(),
        legend.text = element_text(colour = 'black', size = 15, face = 'bold'), legend.box = "horizontal",
        legend.key.width=unit(2, "cm"), legend.key = element_rect(fill='white'))


## function to openfile and extract lengths
process_lengths = function (filename, lib_name) {
	newdf <- read.delim(file=filename, header = FALSE)
	newdf$length <- as.numeric(gsub("^\\@\\w+\\.\\d+\\s{1}length=", "", newdf$V1, perl=TRUE))
	colnames(newdf) <- c("libname", "length")
	newdf$libname <- rep(lib_name, nrow(newdf))
	#cat("Lengths summary", filename, "\n")
	#print(summary(newdf$length))
  newdf
}


```

...     

...  		

Pacbio sequence data for Arabidopsis is published in the following mansucript [http://www.nature.com/articles/sdata201445](http://www.nature.com/articles/sdata201445).

...

I have used Arabidopsis data from P5C3 kits for the following analysis.
Datasets downloaded from [NCBI Sequence Read Archive](http://www.ncbi.nlm.nih.gov/sra/SRX533607[accn])

...

#### box plot of sequence read lengths from pacbio (arabidopsis)

```{r boxplot-pacbio, echo=FALSE, fig.width=12, fig.height=6}
df1 = process_lengths("len_SRR1284093.txt", "lib1")
newdf = process_lengths("len_SRR1284094.txt", "lib2")
df1 <- rbind(df1, newdf)
newdf = process_lengths("len_SRR1284095.txt", "lib3")
df1 <- rbind(df1, newdf)
newdf = process_lengths("len_SRR1284096.txt", "lib4")
df1 <- rbind(df1, newdf)
newdf = process_lengths("len_SRR1284703.txt", "lib5")
df1 <- rbind(df1, newdf)
newdf = process_lengths("len_SRR1284704.txt", "lib6")
df1 <- rbind(df1, newdf)
newdf = process_lengths("len_SRR1284705.txt", "lib7")
df1 <- rbind(df1, newdf)
newdf = process_lengths("len_SRR1284706.txt", "lib8")
df1 <- rbind(df1, newdf)
newdf = process_lengths("len_SRR1284707.txt", "lib9")
df1 <- rbind(df1, newdf)
ggplot(df1, aes(x=libname, y=length, fill=libname)) + geom_boxplot() + scale_y_log10() +
  labs(x ="libraries", y="length in bp", main=NULL) + mytheme + theme(legend.position="none")


```


...


...

#### box plot of scaffold sequence lengths for arabidopsis assemblies (Illumina reads)

```{r boxplot-assembly, echo=FALSE, fig.width=12, fig.height=6}
df1 = read.delim("assembly_lengths/read_lengths_hasty.txt", header=FALSE)
colnames(df1) <- c("assembly", "length")
df1$assembly <- rep("hasty", nrow(df1))
newdf = read.delim("assembly_lengths/read_lengths_soc1.txt", header=FALSE)
colnames(newdf) <- c("assembly", "length")
newdf$assembly <- rep("soc1", nrow(newdf))
df1 <- rbind(df1, newdf)
newdf = read.delim("assembly_lengths/read_lengths_sup1.txt", header=FALSE)
colnames(newdf) <- c("assembly", "length")
newdf$assembly <- rep("sup1", nrow(newdf))
df1 <- rbind(df1, newdf)
newdf = read.delim("assembly_lengths/frags_ordered_lengths.txt", header=FALSE)
colnames(newdf) <- c("assembly", "length")
newdf$assembly <- rep("exponential", nrow(newdf))
df1 <- rbind(df1, newdf)
newdf = read.delim("assembly_lengths/genome_1_lengths.txt", header=TRUE)
colnames(newdf) <- c("assembly", "length")
newdf$assembly <- rep("lognormal_1", nrow(newdf))
df1 <- rbind(df1, newdf)
newdf = read.delim("assembly_lengths/genome_2_lengths.txt", header=TRUE)
colnames(newdf) <- c("assembly", "length")
newdf$assembly <- rep("lognormal_2", nrow(newdf))
df1 <- rbind(df1, newdf)

ggplot(df1, aes(x=assembly, y=length, fill=assembly)) + geom_boxplot() + scale_y_log10() +
  labs(x ="arabidopsis assemblies", y="length in bp", main=NULL) + mytheme + theme(legend.position="none")

```



#### fitting distribtuion to the assembly lengths

```{r fit-distr, echo=FALSE, fig.width=12, fig.height=9}

df1 <- read.delim(file="assembly_lengths/read_lengths_hasty.txt", header = FALSE)
lenden <- density(df1$V2)
contiglen <- data.frame(x=lenden$x, y=lenden$y)

fit1 <- fitdistr(df1$V2, "lognormal")
lnorm_r <- rlnorm(nrow(df1), as.numeric(fit1$estimate['meanlog']), as.numeric(fit1$estimate['sdlog']))
den1 <- density(lnorm_r)
ln_model <- data.frame(x=den1$x, y=den1$y)

fit2 <- fitdistr(df1$V2, "exponential")
expon_r <- rexp(nrow(df1), as.numeric(fit2$estimate['rate']))
den2 <- density(expon_r)
exp_model <- data.frame(x=den2$x, y=den2$y)

fit3 <- fitdistr(df1$V2, "normal")
normal_r <- rnorm(nrow(df1), as.numeric(fit3$estimate['mean']), as.numeric(fit3$estimate['sd']))
den3 <- density(normal_r)
normal_model <- data.frame(x=den3$x, y=den3$y)

ggplot() + geom_line(data=contiglen, aes(x=x, y=y, colour="assembly lengths"), size=1) +
geom_line(data=ln_model, aes(x=x, y=y, colour="lognormal fit"), size=1) + geom_line(data=exp_model, aes(x=x, y=y, colour="exponential fit"), size=1) + geom_line(data=normal_model, aes(x=x, y=y, colour="normal fit"), size=1) + scale_y_sqrt() + mytheme + theme(legend.position=c(0.85,0.9)) + labs(x ="contig lengths", y="density", main=NULL)


```



