### Table of Cotents
1. [April 09 - 10](#april-09th-to-10th)
2. [April 13 - 17](#april-13th-to-17th)
3. [April 20 - 24](#april-20th-to-24th)


## April 09th to 10th

1. Discussion on addition of integer/float to numerator and denominator of a ratio

2. Hypothesis is that higher the number added to each value, lower the amplitude and vice versa

3. This was theoritically tested by [Carlos](https://github.com/calugo) and proved to be as in point 2

4. And was experimentally verified by [Pilar](https://github.com/pilarcormo) for 1 and <1		
[and are available at](https://github.com/pilarcormo/SNP_distribution_method/blob/bed4ed44da7271ea1e0b25a8c9b2693a65ea9d4b/Results/Ratios.md)



## April 13th to 17th

1. Discussed the variant ratio distribution and influence of length in depicting density

2. At the moment ratios are calculated with out representing the length of each genomic fragment.
As we are only testing with in-silico data, it may not influence much, if the contig lengths are uniform. But in reality the contig lengths would vary and needs to be tested.

3. Contigs are also not going to cover the entire genome, so we have to test the influence of missing bits on distribution.

#### To dos:

1. Get de-novo assemblies for various plants and check length distribution

2. Blast de-novo assemblies to completed genome (ex. arabidopsis) and check the contiguity and length distribution

3. Recreate variant ratio distribution for complete genome and de-novo assemblies

4. Use these as models to fit parameters



## April 20th to 24th
