### Table of Cotents
1. [August 10 - 14](#august-10th-to-14th)
2. [August 17 - 21](#august-17th-to-21th)
3. [August 24 - 28](#august-24th-to-28th)


## August 10th to 14th

#### Assemblable part of genome and it's influence on SNP distribution

1. Downloaded genome guided assemblies of Bur-0, C24, Kro-0 and Ler-1 from [1001 genomes](http://1001genomes.org/projects/MPISchneeberger2011/index.html). [Reference](http://www.pnas.org/content/108/25/10249.full)

2. Pacbio de-novo [assembly of Ler-1 using P5-C3 chemistry ](http://blog.pacificbiosciences.com/2014/01/data-release-preliminary-de-novo.html) data is available to [download](http://datasets.pacb.com.s3.amazonaws.com/2014/Arabidopsis/reads/list.html)

3. Denovo assemblies of three rice cultivars [reference] (http://www.genomebiology.com/2014/15/11/506)

4. Idea is to find coverage of assembled genomes against completed reference. Find out regions not covered by these assemblies, exclude variant information from these regions not coverd by the assemblies and check if we still can find the causative mutation or the region of the mutation.

5. Arabidopsis datasets were used to blast against TAIR10 genome and use the information to plot SNP distirbutions and compare between different version of the genomes

6. Hypothesis: Since assembled parts of genomes mostly lack centromeres and repeat regions, SNP distribution may appear cleaner and this would represnt the scenarios observed with denovo analysis of targeted draft genomes

7. plot of homozygous and heterozygous snp distribution with complete and assembled only chromosomes and find the distribution differences.

8. Hypothesis: If plotted with only assemblable parts of the genome, we may not need to create a ratio of the homozygous and heterozygous distributions and may suffice with only homozygous frequencies. Since we are getting rid of high heterozygosity observed in repeat rich regions and centromeres

9. Test this hypothesis by plotting all combinations for total chromosomes and only assmblebel protions of the genome (for arabidopsis and rice at the moment)


## August 17th to 21th

1. Downloaded assemblies for arabidopsis (4 gudied assemblies form 1001 genomes and 1 assembly from pacbio) are blasted against Chromosomes of arabidopsis

2. A file of chromosome covered regions are generated for each assembly using script in mutations_hts_noref/001_blastn_to_gff

3. Chromsome covered regions for each assembly is saved as a gff file in following fodler mutations_hts_noref/002_snps_in_assembled_genome/arabidopsis_assemblies_blast

4. From [Allen et al](http://journal.frontiersin.org/article/10.3389/fpls.2013.00362/full) backcross data variants information generated by [Pilar Corredor] (https://github.com/pilarcormo) was made use to check the assemblable genome effect in finding or missing the causative mutation

5. Data taken from [repo](https://github.com/shyamrallapalli/SNP_distribution_method/tree/master/Reads/BCF2/BCF2_chromosome3)

6. mutations_hts_noref/002_snps_in_assembled_genome/parse_gff_alns_snp_locations.rb script was used to filter out missing variants from assemblies and adjust positions based on assembly size


## August 24th to 28th

1. Plots of snp density along chormosome3 where causative mutation is located are plotted for un-filtered mutant data along with parent and centormere filtered data.

2. Parent filtered variants are used for filtering out missing variants on assembly

3. Plots presented in a [Rmarkdown file] (https://rawgit.com/shyamrallapalli/mutations_hts_noref/master/002_snps_in_assembled_genome/variant_data/BCF2_chromosome3/variant_distribution_report.html) to view

4. SDM is being mock tested using the guided assembly of Ler-1 from 1001 arabidopsis genomes.

5. Sequence read from [Allen et al](http://journal.frontiersin.org/article/10.3389/fpls.2013.00362/full) is used for snp calling and contigs/scaffolds from Ler-1 guided assembly and snp density will be used to test SDM.

6. So far all the test were done using either simulated genomes or artifical contigs generated from single chromosome, so this test will be a first real test of SDM contig arrangement and mutation identification

