### Table of Cotents
1. [August 10 - 14](#august-10th-to-14th)




## August 10th to 14th

#### Assemblable part of genome and it's influence on SNP distribution

1. Downloaded genome guided assemblies of Bur-0, C24, Kro-0 and Ler-1 from [1001 genomes](http://1001genomes.org/projects/MPISchneeberger2011/index.html). [Reference](http://www.pnas.org/content/108/25/10249.full)

2. Pacbio de-novo [assembly of Ler-1 using P5-C3 chemistry ](http://blog.pacificbiosciences.com/2014/01/data-release-preliminary-de-novo.html) data is available to [download](http://datasets.pacb.com.s3.amazonaws.com/2014/Arabidopsis/reads/list.html)

3. Denovo assemblies of three rice cultivars [reference] (http://www.genomebiology.com/2014/15/11/506)

4. Idea is to find coverage of assembled genomes against completed reference. Find out regions not covered by these assemblies, exclude variant information from these regions not coverd by the assemblies and check if we still can find the causative mutation or the region of the mutation.

5. Arabidopsis datasets were used to blast against TAIR10 genome and use the information to plot SNP distirbutions and compare between different version of the genomes

6. Hypothesis: Since assembled parts of genomes mostly lack centromeres and repeat regions, SNP distribution may appear cleaner and this would represnt the scenarios observed with denovo analysis of targeted draft genomes

7. plot of homozygous and heterozygous snp distribution with complete and assembled only chromosomes and find the distribution differences.

8. Hypothesis: If plotted with only assemblable parts of the genome, we may not need to create a ratio of the homozygous and heterozygous distributions and may suffice with only homozygous frequencies. Since we are getting rid of high heterozygosity observed in repeat rich regions and centromeres

9. Test this hypothesis by plotting all combinations for total chromosomes and only assmblebel protions of the genome (for arabidopsis and rice at the moment)
