### Table of Cotents
1. [November 2 - 6]  (#november-2nd-to-6th)


## November 2nd to 6th

1. Variants identified from assembly around the causative mutation are plotted with assembly lengths as bins for sup1, sup2, soc1 and hasty.

2. Distribution of ratios around causative mutation using assembly lengths have not shown any clear pattern of primary peaks close to the causative mutation regardless of whether ratios are adjusted for lengths of the assembly or not.

3. In addition to the assembly, random 1000 iterations of arabidopsis genome fragmented with modelled log-normal distribution derived arabidopsis assemblies were generated to test above points independently. variant frequencies are taken from published studies.

4. Smaller number addtiona (such as 0.1 or 0.01 or lower) as ratio adjustment factor seems to increase ratios dramatically, especially for the contigs/ fragments without heterozygous variants. therefore it is preferred to use anything between 1 and 0.5.

5. From 1000 random iterations of hasty mutant data we can see that on average ~620 contigs had variants out of ~12000 fragments. And only ~33 fragments had higher homozygous than heterozygous variatns. This dataset is a backcross mutant data. 

6. Now generating random data from an outcrossed mutant sequencing and would help us determine the limits of the analysis.