---
output:
  html_document:
    fig_caption: yes
    theme: spacelab
    toc: no
---
# **plots of variant distribution along chromosome length**

Load libraries and the data
```{r setup, include=FALSE}
# load the required libraries and functions
library(knitr)
library(markdown)
#opts_chunk$set(dev = 'pdf')
library(grid)
library(gridExtra)
library(ggplot2)

## cusotom ggplot theme edits
mytheme <- theme(panel.background = element_rect(fill='white', colour='black'),
        text = element_text(size = 20, face='bold', colour='black'),
        axis.title.x = element_text(size = 19, face='bold', color='black', vjust = -1),
        axis.text = element_text(size = 17, face='bold', color='black'),
        panel.grid.major = element_line(linetype = 'dotted', colour='black'),
        panel.border = element_rect(fill=NA, linetype = "solid", colour='black'),
        plot.margin=unit(c(2,2,5,2),'mm'), legend.position=c(0.80,0.90), legend.title = element_blank(),
        legend.text = element_text(colour = 'black', size = 15, face = 'bold'), legend.box = "horizontal",
        legend.key.width=unit(2, "cm"), legend.key = element_rect(fill='white'))

## set working directory
setwd("~/git-hub-repos/shyamrallapalli/mutations_hts_noref/002_snps_in_assembled_genome/contig_snp_distribution/arabidopsis/allen_etal/hsty")

```

...     

...  		

#### 10,000 bp grouping of whole genome

```{r 10kb, echo=FALSE, fig.width=12, fig.height=6}
tenkb <- read.delim(file="per-10000bp-filtered_hsty_mutant.vcf.txt", header = TRUE)
tenkb$numhm <- tenkb$numhm + 0.1
tenkb$numht <- tenkb$numht + 0.1
tenkb$ratio <- tenkb$numhm/tenkb$numht
ggplot(data=tenkb, aes(x=position)) + geom_line(aes(y=ratio/sum(ratio), colour="homozygous to heterozygous ratio")) + 
     geom_line(aes(y=numhm/sum(numhm), colour="homozygous variants")) + 
     geom_line(aes(y=numht/sum(numht), colour="heterozygous variants")) + 
     scale_y_sqrt() + labs(y=NULL, main="genome in 10 KB fragment chunks")

```


...		

...  	


#### 50,000 bp grouping of whole genome

```{r 50kb, echo=FALSE, fig.width=12, fig.height=6}
tenkb <- read.delim(file="per-50000bp-filtered_hsty_mutant.vcf.txt", header = TRUE)
tenkb$numhm <- tenkb$numhm + 0.1
tenkb$numht <- tenkb$numht + 0.1
tenkb$ratio <- tenkb$numhm/tenkb$numht
ggplot(data=tenkb, aes(x=position)) + geom_line(aes(y=ratio/sum(ratio), colour="homozygous to heterozygous ratio")) + 
     geom_line(aes(y=numhm/sum(numhm), colour="homozygous variants")) + 
     geom_line(aes(y=numht/sum(numht), colour="heterozygous variants")) + 
     scale_y_sqrt() + labs(y=NULL, main="genome in 50 KB fragment chunks")

```


...		

...  	


#### 100,000 bp grouping of whole genome

```{r 100kb, echo=FALSE, fig.width=12, fig.height=6}
tenkb <- read.delim(file="per-100000bp-filtered_hsty_mutant.vcf.txt", header = TRUE)
tenkb$numhm <- tenkb$numhm + 0.1
tenkb$numht <- tenkb$numht + 0.1
tenkb$ratio <- tenkb$numhm/tenkb$numht
ggplot(data=tenkb, aes(x=position)) + geom_line(aes(y=ratio/sum(ratio), colour="homozygous to heterozygous ratio")) + 
     geom_line(aes(y=numhm/sum(numhm), colour="homozygous variants")) + 
     geom_line(aes(y=numht/sum(numht), colour="heterozygous variants")) + 
     scale_y_sqrt() + labs(y=NULL, main="genome in 100 KB fragment chunks")

```


...		

...  	


#### 500,000 bp grouping of whole genome

```{r 500kb, echo=FALSE, fig.width=12, fig.height=6}
tenkb <- read.delim(file="per-500000bp-filtered_hsty_mutant.vcf.txt", header = TRUE)
tenkb$numhm <- tenkb$numhm + 0.1
tenkb$numht <- tenkb$numht + 0.1
tenkb$ratio <- tenkb$numhm/tenkb$numht
ggplot(data=tenkb, aes(x=position)) + geom_line(aes(y=ratio/sum(ratio), colour="homozygous to heterozygous ratio")) + 
     geom_line(aes(y=numhm/sum(numhm), colour="homozygous variants")) + 
     geom_line(aes(y=numht/sum(numht), colour="heterozygous variants")) + 
     scale_y_sqrt() + labs(y=NULL, main="genome in 500 KB fragment chunks")

```


...		

...  	


#### 1,000,000 bp grouping of whole genome

```{r 1Mb, echo=FALSE, fig.width=12, fig.height=6}
tenkb <- read.delim(file="per-1000000bp-filtered_hsty_mutant.vcf.txt", header = TRUE)
tenkb$numhm <- tenkb$numhm + 0.1
tenkb$numht <- tenkb$numht + 0.1
tenkb$ratio <- tenkb$numhm/tenkb$numht
ggplot(data=tenkb, aes(x=position)) + geom_line(aes(y=ratio/sum(ratio), colour="homozygous to heterozygous ratio")) + 
     geom_line(aes(y=numhm/sum(numhm), colour="homozygous variants")) + 
     geom_line(aes(y=numht/sum(numht), colour="heterozygous variants")) + 
     scale_y_sqrt() + labs(y=NULL, main="genome in 1 MB fragment chunks")

```

